projects/pastanaga-angular/src/lib/modal/modal.service.ts

```diff
@@ -0,0 +1,83 @@
+import {
+    ApplicationRef,
+    ComponentFactoryResolver,
+    ComponentRef,
+    EmbeddedViewRef,
+    Injectable,
+    Injector,
+    NgZone,
+    Type,
+} from '@angular/core';
+import { IModal } from './base-modal.component';
+import { ModalConfig, ModalRef } from './modal.model';
+import { take } from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class ModalService {
+    hasModalOpened = false;
+
+    private modals: ComponentRef<any>[] = [];
+    private counter = 0;
+
+    constructor(
+        private componentFactoryResolver: ComponentFactoryResolver,
+        private appRef: ApplicationRef,
+        private injector: Injector,
+        private zone: NgZone
+    ) {}
+
+    openModal(component: Type<IModal>, config?: ModalConfig): ModalRef {
```

Comment:
I think I'd prefer a Service that offers opening directly the kind of modal rather than having the caller knowing the component to open...

for example openDialog or openInfo ... it's a mater of taste though...

Reply:
not possible because that would prevent us to keep modal.service agnostic: in pastanaga we have two kinds or modals for now, but anyone can create their own modal in their project and open it using this service.

Reply:
makes sense